apps/challenges/views.py

```diff
@@ -340,3 +358,221 @@ def challenge_phase_split_list(request, challenge_pk):
+                                                     challenge_evaluation_script_file,
+                                                     save=True)
+
+            # Create Leaderboard
+            yaml_file_data_of_leaderboard = yaml_file_data['leaderboard']
+            leaderboard_ids = []
+            for data in yaml_file_data_of_leaderboard:
+                serializer = LeaderboardSerializer(data=data)
+                if serializer.is_valid():
+                    leaderboard = serializer.save()
+                    leaderboard_ids.append(leaderboard.pk)
+
+            # Create Challenge Phase
+            yaml_file_data_of_challenge_phase = yaml_file_data['challenge_phases']
+            challenge_phase_ids = []
+            for data in yaml_file_data_of_challenge_phase:
+                serializer = ChallengePhaseSerializer(data=data, context={'challenge': challenge})
+                if serializer.is_valid():
+                    challenge_phase = serializer.save()
+                    challenge_phase_ids.append(challenge_phase.pk)
+
+                    test_annotation_file = data['test_annotation_file']
+                    if test_annotation_file:
+                        test_annotation_file_path = os.path.join("{}/{}/{}{}".format(BASE_LOCATION,
+                                                                                     unique_folder_name,
+                                                                                     extracted_folder_name,
+                                                                                     test_annotation_file))
+                    if os.path.isfile(test_annotation_file_path):
+                        challenge_test_annotation = open(test_annotation_file_path, 'rb')
+                        challenge_test_annotation_file = File(challenge_test_annotation)
+                        challenge_phase = ChallengePhase.objects.get(pk=challenge_phase.pk)
+                        challenge_phase.test_annotation.save(test_annotation_file,
+                                                             challenge_test_annotation_file,
+                                                             save=True)
+
+            # Create Dataset Splits
+            yaml_file_data_of_dataset_split = yaml_file_data['dataset_splits']
+            dataset_split_ids = []
+            for data in yaml_file_data_of_dataset_split:
+                serializer = DatasetSplitSerializer(data=data)
+                if serializer.is_valid():
+                    dataset_split = serializer.save()
+                    dataset_split_ids.append(dataset_split.pk)
+
+            # Create Challenge Phase Splits
+            yaml_file_data_of_challenge_phase_splits = yaml_file_data['challenge_phase_splits']
+            for data in yaml_file_data_of_challenge_phase_splits:
+                challenge_phase = challenge_phase_ids[data['challenge_phase_id']-1]
```

Comment:
The logic for `-1` here is that since I am storing all the saved `Leaderboards`, `Challenge Phases`, `Dataset splits` in a list and the list indexing starts from 0 and the user will input the fields starting from 1,2 & so on. So, I reduced it by 1 to match the list index with user's input. Please tell if I am not clear.